app/cdap/components/ConnectionList/index.tsx

```diff
@@ -198,76 +280,138 @@ export default function ConnectionList() {
     );
   };
 
+  useEffect(() => {
+    setTabSize(dataForTabs.length);
+  }, [dataForTabs.length]);
+
   let headerContent;
 
   return (
     <Box data-testid="data-sets-parent" className={classes.connectionsListContainer}>
       <SubHeader selectedConnection={dataForTabs[0]?.selectedTab} />
+
       {dataForTabs &&
       Array.isArray(dataForTabs) &&
       dataForTabs.length &&
       dataForTabs[0]?.data?.length > 0 ? (
-        <SelectDatasetWrapper>
-          {dataForTabs.map((each, index) => {
-            const connectionIdRequired =
-              each.data.length && each.data.filter((el) => el.connectionId);
-            if (connectionIdRequired.length) {
-              connectionId = connectionIdRequired[0].connectionId;
-            }
-            if (index === 0) {
-              headerContent = headerForLevelZero();
-            } else {
-              headerContent =
-                refs.current[index]?.offsetWidth < refs.current[index]?.scrollWidth ? (
-                  <CustomTooltip title={dataForTabs[index - 1].selectedTab} arrow>
-                    <Box className={classes.beforeSearchIconClickDisplay}>
-                      <Typography
-                        variant="body2"
-                        className={classes.headerLabel}
-                        ref={(element) => {
-                          refs.current[index] = element;
-                        }}
+        <Box className={classes.connectionsWithInfo}>
+          <SelectDatasetWrapper>
+            {filteredData &&
+              Array.isArray(filteredData) &&
+              filteredData?.map((eachFilteredData, index) => {
+                if (eachFilteredData.data.filter((el) => el.connectionId).length) {
+                  connectionId = eachFilteredData.data.filter((el) => el.connectionId)[0]
+                    .connectionId;
+                }
+                if (index === 0) {
+                  headerContent = headerForLevelZero();
+                } else {
+                  headerContent = (
+                    <>
+                      <Box
+                        className={
+                          eachFilteredData.isSearching
+                            ? classes.hideComponent
+                            : classes.beforeSearchIconClickDisplay
+                        }
+                      >
+                        {headersRefs.current[index]?.offsetWidth <
+                        headersRefs.current[index]?.scrollWidth ? (
+                          <CustomTooltip title={dataForTabs[index - 1].selectedTab} arrow>
+                            <Typography
```

Comment:
here, both if and else return elements are same but for if we just have a wrapper on top of the element
can we refactor to conditionalize the wrapper but the element will be written only once
plz check how to achieve this

Reply:
Done